File: app/src/main/java/com/an/trailers/ui/base/custom/menu/MenuDrawerToggle.java
package com.an.trailers.ui.base.custom.menu;

import android.content.res.Configuration;
import android.view.MenuItem;
import android.view.View;
import android.widget.LinearLayout;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;

import java.util.List;


public abstract class MenuDrawerToggle extends ActionBarDrawerToggle implements ViewAnimator.ViewAnimatorListener {

    private AppCompatActivity activity;
    private ViewAnimator viewAnimator;
    private LinearLayout layoutView;
    public MenuDrawerToggle(AppCompatActivity activity,
                            DrawerLayout drawerLayout,
                            Toolbar toolbar,
                            LinearLayout layoutView,
                            int openDrawerContentDescRes,
                            int closeDrawerContentDescRes,
                            List<SlideMenuItem> slideMenuItems) {

        super(activity, drawerLayout, toolbar, openDrawerContentDescRes, closeDrawerContentDescRes);
        this.activity = activity;
        this.layoutView = layoutView;
        viewAnimator =new ViewAnimator(activity, slideMenuItems, drawerLayout, this);
    }

    @Override
    public void onDrawerSlide(View drawerView, float slideOffset) {
        super.onDrawerSlide(drawerView, slideOffset);
        if (slideOffset > 0.6 && layoutView.getChildCount() == 0) {
            viewAnimator.displayMenuContent();
        }
    }

    @Override
    public void onDrawerOpened(View drawerView) {
        super.onDrawerOpened(drawerView);
    }

    @Override
    public void onDrawerClosed(View drawerView) {
        super.onDrawerClosed(drawerView);
        layoutView.removeAllViews();
        layoutView.invalidate();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (onOptionsItemSelected(item)) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
    }

    @Override
    public void updateHomeButton(boolean enabled) {
        activity.getSupportActionBar().setHomeButtonEnabled(enabled);
    }

    @Override
    public void addViewToContainer(View view) {
        layoutView.addView(view);
    }

    public void onDestroy() {
        activity = null;
    }

    public int getSelectedPosition() {
        return viewAnimator.getSelectedPosition();
    }
}

File: app/src/main/java/com/an/trailers/ui/base/custom/menu/MenuDrawerToggle.kt
package com.an.trailers.ui.base.custom.menu

import android.view.MenuItem
import android.view.View
import android.widget.LinearLayout
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.drawerlayout.widget.DrawerLayout


abstract class MenuDrawerToggle(
    private var activity: AppCompatActivity?,
    drawerLayout: DrawerLayout,
    toolbar: Toolbar,
    private val layoutView: LinearLayout,
    openDrawerContentDescRes: Int,
    closeDrawerContentDescRes: Int,
    slideMenuItems: List<SlideMenuItem>

) : ActionBarDrawerToggle(activity, drawerLayout, toolbar, openDrawerContentDescRes, closeDrawerContentDescRes),
    ViewAnimator.ViewAnimatorListener {

    private var viewAnimator: ViewAnimator = ViewAnimator(
        activity!!, slideMenuItems, drawerLayout, this)

    val selectedPosition: Int
        get() = viewAnimator.selectedPosition

    override fun onDrawerSlide(drawerView: View, slideOffset: Float) {
        super.onDrawerSlide(drawerView, slideOffset)
        if (slideOffset > 0.6 && layoutView.childCount == 0) {
            viewAnimator.displayMenuContent()
        }
    }

    override fun onDrawerClosed(drawerView: View) {
        super.onDrawerClosed(drawerView)
        layoutView.removeAllViews()
        layoutView.invalidate()
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if (onOptionsItemSelected(item)) {
            true
        } else super.onOptionsItemSelected(item)
    }

    override fun updateHomeButton(enabled: Boolean) {
        activity!!.supportActionBar!!.setHomeButtonEnabled(enabled)
    }

    override fun addViewToContainer(view: View) {
        layoutView.addView(view)
    }

    fun onDestroy() {
        activity = null
    }
}
